pkg/sensors/sync.go

```diff
@@ -276,6 +276,10 @@ func UnloadSensor(ctx context.Context, bpfDir, mapDir string, sensor *Sensor) er
 		return fmt.Errorf("unload of sensor %s failed: sensor not loaded", sensor.Name)
 	}
 
+	if sensor.UnloadHook != nil {
```

Comment:
Note that there is a potential typed `nil` issue here depending on how `sensor.UnloadHook` is setup.

Reply:
sensor.UnloadHook is defined in type Sensor struct in sensors.go, as a SensorUnloadHook interface containing the Unload() implementation. So, IIRC, whatever is assigned to sensor.UnloadHook has to include an implementation of Unload(), so if not nil then the implementation exists, otherwise is nil, and doesn't have a type. It seems to work correctly where some sensors define it and others don't. Can you describe the case where it would go wrong so I can tighten this up if necessary? Thanks!

Reply:
> […] so if not nil then the implementation exists, otherwise is nil, and doesn't have a type. […]

This is not completely accurate, an interface variable (such as `Sensor.UnloadHook`) is basically a fat pointer with both type and value. Here is a simple example showcasing the issue that can arise:

```go
package main

import "fmt"

type SensorUnloadHook interface {
	Unload() error
}

type Sensor struct {
	UnloadHook SensorUnloadHook
}

type HookImpl struct{}

// Unload implement SensorUnloadHook for HookImpl.
func (*HookImpl) Unload() error {
	fmt.Println("HookImpl.Unload")
	return nil
}

func NewSensor(onUnload SensorUnloadHook) *Sensor {
	return &Sensor{
		UnloadHook: onUnload,
	}
}

func main() {
	ok := NewSensor(nil)
	// prints true
	fmt.Println(ok.UnloadHook == nil)

	var h *HookImpl = nil
	boom := NewSensor(h)
	// prints false!
	fmt.Println(boom.UnloadHook == nil)
}

```

Reply:
I see. If you explicitly assign a typed nil then that will of course fail. Is there a better way to build in an optional interface?

Reply:
AFAIK there are several potential  fix / workaround:

1. declare the unload hook as a function instead of an interface. This would sidestep the problem entirely by always using a concrete type.
2. put the burden on the interface implementations and require them to be `nil`-safe. In my example above, you can actually call `boom.UnloadHook.Unload()` because it doesn't use the `nil` receiver.
3. use `reflect` to dynamically find whether we have a typed `nil`,  e.g. `sensor.UnloadHook != nil && reflect.ValueOf(sensor.UnloadHook) != nil`

Pick you poison :sweat_smile:

Reply:
Thanks! Wasn't aware I could just have a function pointer and that it could be nil, so (1) above is actually the best solution to this. I've updated the commit.